packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -878,6 +906,39 @@ export const useAzureCommunicationCallAdapter = (
   return adapter;
 };
 
+const createAzureCommunicationCallAdapterFromClientStable = async (
+  callClient: StatefulCallClient,
+  callAgent: CallAgent,
+  locator: CallAdapterLocator
+): Promise<CallAdapter> => {
+  const deviceManager = (await callClient.getDeviceManager()) as StatefulDeviceManager;
+  return new AzureCommunicationCallAdapter(callClient, locator, callAgent, deviceManager);
+};
+
+let createAzureCommunicationCallAdapterFromClientToExport: (
+  callClient: StatefulCallClient,
+  callAgent: CallAgent,
+  locator: CallAdapterLocator,
+  /* @conditional-compile-remove(rooms) */ roleHint?: Role
+) => Promise<CallAdapter> = createAzureCommunicationCallAdapterFromClientStable;
+
+/* @conditional-compile-remove(rooms) */
+/**
+ * Create a {@link CallAdapter} using the provided {@link StatefulCallClient} and a provided role
+ * to hint the user of their role specific for Rooms calls.
+ *
+ * @beta
+ */
+createAzureCommunicationCallAdapterFromClientToExport = async (
```

Comment:
I don't understand why you had to introduce `createAzureCommunicationCallAdapterFromClientToExport` in this way.

Can't you conditionally compile just the argument to `createAzureCommunicationCallAdapterFromClient` along with the implementation?

Reply:
This is a hack I had to do because if I cinditionally compile the just the function argument like so:
![image](https://user-images.githubusercontent.com/79475487/200110664-f6ba08d3-5a2e-4f10-9ada-194145f1f5f0.png)
The stable api changes includes the options argument like so:
![image](https://user-images.githubusercontent.com/79475487/200110674-d01f5e92-20bc-4fe9-94a7-2b6ea960560f.png)

Reply:
hmm, @prprabhu-ms @PorterNan @anjulgarg is there any change we can make to the babel config to support this?

Reply:
Hmm, what it would be like in preprocessed file? Wondering why the hack would work

Reply:
It looks much better now without the let variable reassignment 👍